sbi/simulators/simutils.py

```diff
@@ -134,6 +135,79 @@ def simulation_wrapper(
     return parameters, observations
 
 
+def simulation_wrapper_batch(
+        simulator: Callable,
+        parameter_sample_fn: Callable,
+        num_samples: int,
+        simulation_batch_size: int):
+    """
+    Runs simulations in a loop. Each loop simulates simulation_batch_size parameter sets until
+    a total of num_samples is reached
+
+    Args:
+        simulator: Simulator function.
+            If simulation_batch_size == 1: takes in thetas of shape (num_dim_theta)
+                and outputs xs of shape (num_dim_x)
+            If simulation_batch_size > 1: takes in thetas of shape (simulation_batch_size, num_dim_theta)
+                and outputs xs of shape (simulation_batch_size, num_dim_x)
+        parameter_sample_fn: Function to call for generating theta, e.g. prior sampling
+        num_samples: Number of simulations to run
+        simulation_batch_size: Number of simulations that are run within a single batch
+            If simulation_batch_size == -1, we run a batch with all simulations required,
+            i.e. simulation_batch_size = num_samples
+
+    Returns: torch.Tensor parameters theta, torch.Tensor simulated data x
+    """
+
+    # todo: use function in SNL and SRE
+
+    # generate theta by sampling from prior (round 1) or proposal (round > 1)
+    parameters = parameter_sample_fn(num_samples)
+
+    # if simulation_batch_size == -1, we run all simulations in a single batch
+    if simulation_batch_size == -1:
```

Comment:
I'd add a more succint comment just below the if. Also, I am curious why -1 instead of None.
`# single batch`